packages/common/src/compiler/filtersCompiler.ts

```diff
@@ -23,38 +23,40 @@ export const renderStringFilterSql = (
     filter: FilterRule,
 ): string => {
     const filterType = filter.operator;
+    const escapedFilterValues = filter.values?.map((v) =>
+        typeof v === 'string' ? v.replace(/'/g, "''") : v,
```

Comment:
I think this solution might depend on the warehouse... each warehouse has different quote / escaping characters...

Reply:
Yeah, I suggested knex approach above

Reply:
knex option would imply installing all the adapters. I really think we will have to change our architecture so each warehouse client can build the SQL independently (CC: @owlas). As for a simple fix, I wouldn't mind this fix, seems like the double quote escape is pretty standard SQL but we should test it. 
Are there any more special charts we need to escape?

Reply:
The safest way to do this for any database would be to use query parameters provided by the underlying query drivers (e.g. `query('select * from T where x = ?', [userInput])` so `userInput` is always escaped by the db driver). But that would involve the change you suggested @ZeRego - we need to template per warehouse.

We already have some of that at run time, and we switch the quote char depending on the database (`queryBuilder.ts`)

Reply:
yep, I agree on all mentioned above. this is the simplest snippet `knex.raw('?', value)` that outputs a string (locally without running an sql) but it needs a warehouse config in order to work. meanwhile, in the sake of the linked issue I made some refactoring to make this PR acceptable